internal/server/singleprocess/service_trigger.go

```diff
@@ -71,3 +76,125 @@ func (s *service) ListTriggers(
 
 	return &pb.ListTriggerResponse{Triggers: result}, nil
 }
+
+func (s *service) RunTrigger(
+	ctx context.Context,
+	req *pb.RunTriggerRequest,
+) (*pb.RunTriggerResponse, error) {
+	if err := serverptypes.ValidateRunTriggerRequest(req); err != nil {
+		return nil, err
+	}
+	log := hclog.FromContext(ctx)
+
+	runTrigger, err := s.state.TriggerGet(req.Ref)
+	if err != nil {
+		return nil, err
+	}
+
+	log = log.With("run_trigger", runTrigger.Id)
+
+	log.Debug("building run trigger job")
+
+	// Build the job(s)
+	job := &pb.Job{
+		Workspace: runTrigger.Workspace,
+		Labels:    map[string]string{"trigger/id": runTrigger.Id},
+	}
+
+	// TODO(briancain): is there an easy way to convert this without the big switch
+	// so that we don't have to add new case statements any time a new operation is supported?
```

Comment:
Probably, but depends how ugly you want to get lol. This is probably good for now.